### DISCUSSION TITLE
Multiple Outputs in sync to pipe and speakers?

### DISCUSSION THREAD

**Discussion by chrisgott** [mikebrady/shairport-sync#1949] [Ideas]
Hello, I wanted to kindly ask if it’s possible to output simultaneously to a pipe and a regular output at the same time. It seems I’m not the only one interested in running a light show (using WLED or similar) in parallel with a regular sound card, all while using the same machine (or even streaming it to a WLED device).

**Comment by mikebrady** [comment]
Thanks for the question. Unfortunately, the answer at present is “no”, I’m afraid — Shairport Sync is set up to play to one backend only.

But I’ll put it on the list! It may not be too difficult to add it on in a slightly different form to a full backend.

**Comment by mikebrady** [comment]
@chrisgott, I've been thinking a bit about this, and it strikes me that it would be nice, from Shairport Sync's POV, to add a little metadata to the audio feed. Rather than being just a continuous sequence of audio samples, it would have to be a sequence of frames containing a few fields, e.g. rate, channel count, format, some kind of timing information, followed by, typically, 352 or 1024 audio frames. Would that cause you guys a lot of grief, i.e. having to strip off the metadata from each frame?
